features/steps/data.py

```
@@ -46,6 +48,48 @@ def step_impl(context):
         raise context.exception
 
 
+@then("I read the csv output '{csv_output}'")
+def step_impl(context, csv_output):
+    try:
+        context.csv_output = pd.read_csv(csv_output)
+
+    except Exception as exc:
+        raise exc
+
+
+@then("the csv output should have '{number}' rows")
+def step_impl(context, number):
+    num_rows = len(context.csv_output.index)
+
+    assert num_rows == int(number)
+
+
+@then("the csv output has the columns")
+def step_impl(context):
+
+    for column in context.table:
+        assert column in context.csv_output.columns
+
+    
+@then("I read the metadata output '{metadata_output}'")
+def step_impl(context, metadata_output):
+    try:
+        json_output_path = Path(metadata_output)
+
+        metadata_file = open(json_output_path)
+
+        context.json_output = json.load(metadata_file)
+
+    except Exception as exc:
```

no need to catch and raise, just let it raise, see other comment